### PR TITLE
[FEATURE] Explorer is sharable

### PR DESCRIPTION
**Pull request by Gladorme**
# Description

Explorer is sharable! When you edit something it will automatically update query params in the url 😄 

# Screenshots

image

# Checklist

- [x] Pull request has a descriptive title and context useful to a reviewer.
- [x] Pull request title follows the `[<catalog_entry>] <commit message>` naming convention using one of the
  following `catalog_entry` values: `FEATURE`, `ENHANCEMENT`, `BUGFIX`, `BREAKINGCHANGE`, `DOC`,`IGNORE`.
- [x] All commits have [DCO signoffs](https://github.com/probot/dco#how-it-works).

## UI Changes

- [x] Changes that impact the UI include screenshots and/or screencasts of the relevant changes.
- [x] Code follows the [UI guidelines](https://github.com/perses/perses/blob/main/ui/ui-guidelines.md).
- [x] Visual tests are stable and unlikely to be flaky.
  See [Storybook](https://github.com/perses/perses/tree/main/ui/storybook#visual-tests)
  and [e2e](https://github.com/perses/perses/tree/main/ui/e2e#visual-tests) docs for more details. Common issues
  include:
  - Is the data inconsistent? You need to mock API requests.
  - Does the time change? You need to use consistent time values or mock time utilities.
  - Does it have loading states? You need to wait for loading to complete.

### CODE REVIEW

**Comment by Gladorme** [comment]
Need to fix an issue when switching tab 😞

File: ui/explore/src/components/ExploreManager/ExplorerManagerProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    <ExplorerManagerContext.Provider value={{ explorer, tab, queries, setExplorer, setTab, setQueries }}>
+      {children}
+    </ExplorerManagerContext.Provider>
+  );
+}
+
+export function useExplorerManagerContext(): ExplorerManagerContextType {
+  const ctx = useContext(ExplorerManagerContext);
+  if (ctx === undefined) {
+    throw new Error('No ConfigContext found. Did you forget a Provider?');
```
Comment (celian-garcia) [inline ui/explore/src/components/ExploreManager/ExplorerManagerProvider.tsx:77]: No ExplorerManagerContext ?

File: ui/explore/src/components/ExploreManager/ExplorerManagerProviderWithQueryParams.tsx [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import React, { ReactNode } from 'react';
+
+import { JsonParam, NumberParam, useQueryParams } from 'use-query-params';
+import { QueryDefinition } from '@perses-dev/core';
+import { ExplorerManagerProvider } from './ExplorerManagerProvider';
+
+const exploreQueryConfig = {
+  explorer: NumberParam,
+  tab: NumberParam,
+  queries: JsonParam,
```
Comment (celian-garcia) [inline ui/explore/src/components/ExploreManager/ExplorerManagerProviderWithQueryParams.tsx:23]: There's no array type ? :(

**Review by celian-garcia — APPROVED** [review]
Minor changes but overall that looks pretty clean ! thank you for that amazing feature and congrats for the 2000th pull requests !! :partying_face: :balloon:

File: ui/explore/src/components/ExploreManager/ExplorerManagerProviderWithQueryParams.tsx (hunk 2 above)
Comment (Gladorme) [reply ui/explore/src/components/ExploreManager/ExplorerManagerProviderWithQueryParams.tsx:23]: Only for string :/

**Pull request merged**